lib/unparser/writer/array.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Unparser
+  module Writer
+    class Array
+      include Writer, Adamantium
+
+      MAP = {
+        dsym: '%I',
+        sym:  '%i',
+        dstr: '%W',
+        str:  '%w'
+      }.freeze
+      private_constant(*constants(false))
+
+      def emit_compact # rubocop:disable Metrics/AbcSize
+        children_generic_type = array_elements_generic_type || :sym
+
+        write(MAP.fetch(children_generic_type))
+
+        # NOTE: I'd like to use stricter `children.fetch(0)` here instead of `children.first`,
```

Comment:
@viralpraxis 

Can you create an `Unparser::Util` module and port that one:

https://github.com/mbj/mutant/blob/9029cbfaa91bc6f9692112766672f36e7e4bd5ac/lib/mutant/util.rb#L9-L21

Than switch call sides to it? Specs are trivial to copy from mutant also.

Reply:
Yep, done 👍🏻

Reply:
>    Error raised by `Util.one` if size is **less than zero** or greater than one
 >   SizeError = Class.new(IndexError)

:-)

Reply:
brain fart, no need to port that doc error :P